src/mqm/geo_process.py

```diff
@@ -211,14 +212,22 @@ def bounding_box_process(self):
         
         # loop through all geojson files
         for f in os.listdir(self.folder_path):
+            print(f)
             # load the Geo-json file and ignore other files
-            if os.path.splitext(os.path.join(self.folder_path, f))[1] == '.geojson':
+            if ('geojson' in (os.path.join(self.folder_path, f)).split('.')) and not f.startswith('.'):
                 if len(os.path.splitext(f)[0].split('-')) == 3:    # pull out this function
-                    name_num_list.append([os.path.splitext(f)[0].split('-')[0], int(os.path.splitext(f)[0].split('-')[2])])
+                    if os.path.splitext(os.path.join(self.folder_path, f))[1] == '.gz':
+                        name_num_list.append([os.path.splitext(f)[0].split('-')[0], int((os.path.splitext(f)[0].split('-')[2]).split('.')[0])])
+                    else:
+                        name_num_list.append([os.path.splitext(f)[0].split('-')[0], (int(os.path.splitext(f)[0].split('-')[2]))])
+                print(name_num_list)
 
                 # open geojson files
-                with open(os.path.join(self.folder_path, f), encoding='utf-8') as new_f:
-                    data = json.load(new_f)
+                if os.path.splitext(os.path.join(self.folder_path, f))[1] == '.gz':
```

Comment:
Is there a way to recursively fetch all geojson files so we don't require a nested input?

Reply:
I can implement os.walk here to recursively grab each GeoJSON in each countrified subdirectory, but the subdirectories are passed in via the folder_path variable in the main function. Does 'nested input' refer to the current requirement of having a different subdirectory for input area rather than only running mqm for a set of files in the immediate directory, or does it refer to recursively searching within each subdirectory even if there are additional subdirectories within them?

Reply:
Sorry for the confusion! I was referring to mqm requiring you have a input folder structure of:
```
/data
 /subdata
   filename.geojson
```
I'm hoping we can update it to also handle:
```
/data
 mydata.geojson
```

Reply:
I dug into this a bit and found that this functionality is already handled in the main function via:

`    folder_list = util.get_sub_directionaries(input_folder)
    if len(folder_list) == 0:
        folder_list.append(input_folder)`

Currently, the tool checks for nested directories, and uses subdirectories if that is the case. If it can't find any subdirectories, it instead updates the list that is later iterated over by appending the current input directory. This covers the two cases listed above, but won't cover a case like this:

`/data
    filename1.geojson
    /subdirectory
        filename2.geojson`

However, the above case is probably not too important for most users. Because this is already in place, it might make more sense to revert back the addition of os.walk, because although we can now handle some edge cases like this:

`/data
    /subdata
        /atlas-checks-geojsons (however many levels down)
            filename.geojson
        boundary.json`

We can't do the same if the same format exists in the base directory because it will assume that the boundary file is housed in the subdirectory. This is only an issue if the user decides to pass in the atlas check results in a folder, but the boundary file separately, which is probably not a common case either. It may be better to still require that the atlas checks and boundary share the same directory to ensure consistency between each case.

Reply:
Yeah, I think it makes sense to require the boundary file to be located in the same directory.